hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/DeploymentHelper.java

```diff
@@ -109,20 +111,30 @@ public static void successCancellation(final JpaAction action, final ActionRepos
      *
      * @return the result of the action
      */
+    @SuppressWarnings({ "squid:S2221", /* for catching the generic Exception instead of specific ones */
+                        "squid:S1162" /* for throwing a checked exception */ })
     public static <T> T runInNewTransaction(@NotNull final PlatformTransactionManager txManager,
-            final String transactionName, final int isolationLevel, @NotNull final TransactionCallback<T> action) {
+            final String transactionName, final int isolationLevel, @NotNull final TransactionCallback<T> action) throws
+            TransactionExecutionException {
         final DefaultTransactionDefinition def = new DefaultTransactionDefinition();
         def.setName(transactionName);
         def.setReadOnly(false);
         def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
         def.setIsolationLevel(isolationLevel);
-        return new TransactionTemplate(txManager, def).execute(action);
+        final TransactionTemplate transactionTemplate = new TransactionTemplate(txManager, def);
+
+        try {
+            return transactionTemplate.execute(action);
+        } catch (final Exception e) {
+            LOG.error("Caught exception during transaction execution, transactionName: {}!", transactionName, e);
```

Comment:
You don't need to log and rethrow the exception, I would just extend the message of TransactionExecutionException with transaction name